controller.go

```diff
@@ -68,8 +72,17 @@ func (c *Controller) Run(ctx context.Context) (err error) {
 		}
 	}()
 
+	if _, err = labels.Parse(c.IngressSelector) ; err != nil {
+		return fmt.Errorf("Invalid Ingress selector - %s", err)
```

Comment:
Please add dependency on https://github.com/pkg/errors and use `errors.Wrap(err, "parse ingress selector failed")`

Reply:
Done. I see there's another return fmt.error "for could not sync cache", but I wasn't sure if you wanted that wrapped as well, so I left it alone.